pkg/controller/authconfig/authconfig.go

```diff
@@ -0,0 +1,299 @@
+	configv1client "github.com/openshift/client-go/config/clientset/versioned/typed/config/v1"
+	routeclient "github.com/openshift/client-go/route/clientset/versioned/typed/route/v1"
+	"github.com/openshift/cluster-authentication-operator/pkg/api"
+	"github.com/openshift/cluster-authentication-operator/pkg/utils"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
+)
+
+var kasServicePort int
+
+func init() {
+	var err error
+	kasServicePort, err = strconv.Atoi(os.Getenv("KUBERNETES_SERVICE_PORT_HTTPS"))
+	if err != nil {
+		klog.Infof("defaulting KAS service port to 443 due to parsing error: %v", err)
+		kasServicePort = 443
+	}
+}
+
+const (
+	authenticationConfigControllerWorkQueueKey = "key"
+)
+
+type AuthenticationConfigController struct {
+	operatorClient *api.OperatorClient
+	authentication configv1client.AuthenticationInterface
+	services       corev1client.ServicesGetter
+	endpoints      corev1client.EndpointsGetter
+	route          routeclient.RouteInterface
+	queue          workqueue.RateLimitingInterface
+}
+
+func NewAuthenticationConfigController(
+	configClient configclient.Interface,
+	kubeClient kubernetes.Interface,
+	routeClient routeclient.RouteV1Interface,
+	operatorClient *api.OperatorClient) *AuthenticationConfigController {
+	return &AuthenticationConfigController{
+		operatorClient: operatorClient,
+		authentication: configClient.ConfigV1().Authentications(),
+		route:          routeClient.Routes("openshift-authentication"),
+		services:       kubeClient.CoreV1(),
+		endpoints:      kubeClient.CoreV1(),
+		queue:          workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "IngressStateController"),
+	}
+}
+
+func (c *AuthenticationConfigController) Run(workers int, stopCh <-chan struct{}) {
+	defer utilruntime.HandleCrash()
+	defer c.queue.ShutDown()
+
+	klog.Infof("Starting AuthenticationConfigController")
+	defer klog.Infof("Shutting down AuthenticationConfigController")
+
+	// doesn't matter what workers say, only start one.
+	go wait.Until(c.runWorker, time.Second, stopCh)
+
+	// add time based trigger
+	go wait.Until(func() { c.queue.Add(authenticationConfigControllerWorkQueueKey) }, time.Minute, stopCh)
+
+	<-stopCh
+}
+
+func (c *AuthenticationConfigController) runWorker() {
+	for c.processNextWorkItem() {
+	}
+}
+
+func (c *AuthenticationConfigController) processNextWorkItem() bool {
+	dsKey, quit := c.queue.Get()
+	if quit {
+		return false
+	}
+	defer c.queue.Done(dsKey)
+
+	err := c.sync()
+	if err == nil {
+		c.queue.Forget(dsKey)
+		return true
+	}
+
+	utilruntime.HandleError(fmt.Errorf("%v failed with : %v", dsKey, err))
+	c.queue.AddRateLimited(dsKey)
+
+	return true
+}
+
+func (c *AuthenticationConfigController) sync() error {
```

Comment:
is this something for @mfojtik's controller factory?

Reply:
I update the PR now to use controller factory